### PR TITLE
feat(components): connect components

### PR DESCRIPTION
**Pull request by brandonyeoxg**
scope:
* connects repo, account service and api service together
* update docker-compsoe yaml to include db component
* add postman collection for exploratory testing
* update gitignore

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -8,26 +8,45 @@ import (
...
 	runRouter(r)
 }
 
+func setupAccountService() account.AccountService {
+	accPgSettings := postgres.ConnSettings{
+		Host:     "avocadoro-db",
+		Port:     5432,
+		DBName:   "avocadoro",
+		User:     "postgres",
+		Password: "postgres123",
```
Comment (deborahlow97) [inline main.go:32]: Hey dont put credentials out there even dummy ones

File: main.go [hunk 2]
```diff
@@ -8,26 +8,45 @@ import (
...
+	}
+	accRepo, err := postgres.NewRepository(accPgSettings)
+	if err != nil {
+		fmt.Printf("Error connecting to DB. Cause: %s", err)
```
Comment (deborahlow97) [inline main.go]: use log instead of printf

File: testfiles/avocadoro.postman_collection.json [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+			},
+			"response": []
+		},
+		{
+			"name": "accounts - get",
+			"request": {
+				"method": "GET",
+				"header": [],
+				"url": {
+					"raw": "{{url}}/accounts",
```
Comment (deborahlow97) [inline testfiles/avocadoro.postman_collection.json:31]: wouldn't `account` make more sense than `accounts` in this case

File: testfiles/avocadoro.postman_collection.json [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+					"host": [
+						"{{url}}"
+					],
+					"path": [
+						"accounts"
+					]
+				}
+			},
+			"response": []
```
Comment (deborahlow97) [inline testfiles/avocadoro.postman_collection.json:40]: is the response gonna be fleshed in this pr

File: testfiles/avocadoro.postman_collection.json [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		},
+		{
+			"name": "accounts - delete",
```
Comment (deborahlow97) [inline testfiles/avocadoro.postman_collection.json:60]: why do we have an option for user to delete their account ah? isit only for our use

File: docker-compose.dev.yml [hunk 6]
```diff
@@ -6,3 +6,28 @@ services:
...
+    volumes:
+      - .:/app
+
+  postgres:
+    image: postgres:11-alpine
+    container_name: avocadoro-db
+    restart: "unless-stopped"
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres123
```
Comment (deborahlow97) [inline docker-compose.dev.yml:20]: the credentials out there again

File: main.go [hunk 7]
```diff
@@ -8,26 +8,45 @@ import (
...
+		return nil
+	}
+	return account.NewAccountService(accRepo)
+}
+
 func setupRouter() *chi.Mux {
 	r := chi.NewRouter()
```
Comment (deborahlow97) [inline main.go]: can use something more descriptive instead of `r` or `h` or `c` within this file

**Review by deborahlow97 — CHANGES_REQUESTED** [review]

File: main.go [hunk 8]
```diff
@@ -8,26 +8,45 @@ import (
...
 	r.Use(middleware.Logger)
 	return r
 }
 
-func initRoutes(r *chi.Mux) {
-	h := api.NewHandler(nil)
+func initRoutes(r *chi.Mux, svc account.AccountService) {
```
Comment (deborahlow97) [inline main.go]: dont use `r` here

**Review by deborahlow97 — CHANGES_REQUESTED** [review]

File: docker-compose.dev.yml (hunk 6 above)
Comment (brandonyeoxg) [reply docker-compose.dev.yml:20]: This is for the mock database for dev, so there should not be any security concern.

File: testfiles/avocadoro.postman_collection.json (hunk 3 above)
Comment (brandonyeoxg) [reply testfiles/avocadoro.postman_collection.json:31]: In rest api design we use nouns in plural form for endpoints.

https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/

File: testfiles/avocadoro.postman_collection.json (hunk 4 above)
Comment (brandonyeoxg) [reply testfiles/avocadoro.postman_collection.json:40]: Meant to keep this empty to make it clean.

File: testfiles/avocadoro.postman_collection.json (hunk 5 above)
Comment (brandonyeoxg) [reply testfiles/avocadoro.postman_collection.json:60]: We can use this internally.

Comment (deborahlow97) [reply testfiles/avocadoro.postman_collection.json:60]: okie

**Review by deborahlow97 — APPROVED** [review]
LGTM 🌊

**Pull request merged**